engine/engine.go

```diff
@@ -317,6 +317,10 @@ func (p *Engine) ReceiveSubject(ctx context.Context, providerID, subjectID, call
 		newStatus = alert.StatusActive
 	case notification.ResultResolve:
 		newStatus = alert.StatusClosed
+	case notification.ResultEscalate:
+		newStatus = alert.StatusTriggered
+		alertIDS := []int{cb.AlertID}
+		p.a.EscalateMany(ctx, alertIDS)
```

Comment:
Performing the escalation and returning here may make more sense since the code below is for status updates.

The escalation action will handle updating the status automatically elsewhere.

Reply:
@mastercactapus thanks for the comments. I have updated the code. Since we escalate only one alert at a time, I have assigned cb.AlertID the updated alert id (alertIds[0]).